src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\TypedReference;
+use Symfony\Contracts\Service\Attribute\SubscribedService;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class AutowireIterator extends Autowire
+{
+    /**
+     * @see ServiceSubscriberInterface::getSubscribedServices()
+     *
+     * @param array<string|SubscribedService>|string $services An explicit list of services or a tag name
+     */
+    public function __construct(
+        array|string $services,
+        string $indexAttribute = null,
+        string $defaultIndexMethod = null,
+        string $defaultPriorityMethod = null,
+        string|array $exclude = [],
+        bool $excludeSelf = true,
+    ) {
+        if (\is_string($services)) {
+            parent::__construct(new TaggedIteratorArgument($services, $indexAttribute, $defaultIndexMethod, false, $defaultPriorityMethod, (array) $exclude, $excludeSelf));
+
+            return;
+        }
+
+        $references = [];
+
+        foreach ($services as $key => $type) {
+            $attributes = [];
+
+            if ($type instanceof SubscribedService) {
+                $key = $type->key ?? $key;
```

Comment:
I added to `?? $key` here but I'm not sure how this should work? Should the `$services` array key take precedence over the `SubscribedService`'s key?

Reply:
Why not, but we have to sync with https://github.com/symfony/symfony/blob/9767270b34e97d1bbf340f9ab3358a0178dccb8a/src/Symfony/Component/DependencyInjection/Compiler/RegisterServiceSubscribersPass.php#L82
No opinion yet, we need to think about why we didn't back then and why we would now

Reply:
Hmm, yeah. I think it's safe to propagate this change to `RegisterServiceSubscriberPass` so _the key in `SubscribedService` takes precedence over the array key_ (the opposite of what I proposed above).

Reply:
All good for me _as-is_.